CONTRIBUTING.md

```diff
@@ -1,7 +1,73 @@
+# Rationale
+
+Contributions are always welcome. We accept [bug fixes](#bugs),
+[proposals and new ideas](#proposal), or simply [editorial
+improvements](#editorial-improvements).
+
+Please not that by submitting a pull request you are implicitly
+adhering to the [contributions greement](#contributions-agreement)
+
+## Bugs
+
+You can report a bug by creating an
+[issue](https://github.com/ARM-software/acle/issues). You cal also fix
+it yourself with a [pull
+request](https://github.com/ARM-software/acle/pulls).
+
+## Proposals
+
+We are open to any proposal, including the introduction of new sets of
+intrinsics. We _do not_ require to assign copyright to Arm for the
+introduction of new parts of the specifications. The original
+contributor will retain the copyright.
+
+For adding new intrinsics or preprocessor macros, we recommend
+presenting them in a [pull
+request](https://github.com/ARM-software/acle/pulls), together with
+the following information:
+
+1. A rationale covering why these intrinsics or macros are needed.
+2. A design document to keep track of the reasoning that made the
+   proposal reach its current state.
+3. Some consensus gathering in the ecosystem.
```

Comment:
I think (3) could be expressed more concretely -- it could be clearer what is being asked for to meet this bar.